### ISSUE TITLE
TODO: Deprecate/Remove Plugin from Garden

### ISSUE THREAD

**Issue by methylDragon** [gazebosim/gz-common#349]
From: https://github.com/ignitionrobotics/ign-common/pull/348#discussion_r863354402

The `Plugin` files are deprecated from Garden, so we should be removing them.

**Comment by chapulina** [comment]
I started adding deprecation warnings to plugins a while back and lost that branch 🙃 I'll go ahead and do that now.

> so we should be removing them.

This is referring to the `gz/*Plugin*.hh` header files. Fortress (`common4`) users should migrate from common plugins  (`ignition/*Plugin*.hh`) to the [gz-plugin](https://github.com/ignitionrobotics/ign-plugin/) library (`gz/plugin/*.hh`), so we don't need to provide new `gz/*Pluign*.hh` headers.

---

Note that deprecating (tick-tocking) the plugins is an alternative to the proposal in #336 , which I believe we decided against.
